### ISSUE TITLE
Gitlab - SentryCLI failed to set commits

### ISSUE THREAD

**Issue by djw27** [getsentry/sentry-netlify-build-plugin#21]
When using this plugin with a Gitlab integration, we run in to the following error:

```
12:51:09 PM:   cmd: '/opt/buildhome/.netlify-build-plugins/node_modules/@sentry/cli/sentry-cli releases set-commits ac28e1c49ac9312242e09696c2046c53c3926835 --commit theworkshop/web-app@ac28e1c49ac9312242e09696c2046c53c3926835'
12:51:09 PM: }
12:51:09 PM: ​
12:51:09 PM: ┌──────────────────────────────────────────────┐
12:51:09 PM: │ Plugin "@sentry/netlify-build-plugin" failed │
12:51:09 PM: └──────────────────────────────────────────────┘
12:51:09 PM: ​
12:51:09 PM:   Error message
12:51:09 PM:   Error: SentryCLI failed to set commits. You likely need to set up a repository or repository integration.
12:51:09 PM:            Read more: https://docs.sentry.io/workflow/releases/?platform=python#install-repo-integration
```

When running the command locally to replicate the issue, we get a bit more information around why this is failing:

```
$ sentry-cli releases set-commits ac28e1c49ac9312242e09696c2046c53c3926835 --commit theworkshop/web-app@ac28e1c49ac9312242e09696c2046c53c3926835

error: Unknown repo 'theworkshop/web-app'
```

The error we're running into seems to be [due to how whitespace is handled in the repository name with the Gitlab integration](https://help.sentry.io/hc/en-us/articles/360040167333-GitLab-Why-am-I-seeing-an-Invalid-Repository-Names-error-)

Modifying the command locally to the following fixes the issue:

```
$ sentry-cli releases set-commits ac28e1c49ac9312242e09696c2046c53c3926835 --commit theworkshop\ /\ web-app@ac28e1c49ac9312242e09696c2046c53c3926835
```

It should also probably be documented somewhere that the what Sentry uses as 'name' is actually configurable on Gitlab - i.e. if a user has modified their repo name with Gitlab, then this script won't work as this script actually uses the repo 'slug' rather than name for matching.

**Comment by djw27** [comment]
Just to add a comment on a potential 'fix' until I have chance to look at this in any more detail.

https://github.com/getsentry/sentry-netlify-build-plugin/blob/master/index.js#L110 could be edited to the following, you'd probably want a bit more of a definitive check however:

```
const repositoryParts = process.env.REPOSITORY_URL.split(/[:/]/).slice(-2)

const repository = process.env.REPOSITORY_URL.includes('gitlab') ? repositoryParts.join(' / ') : repositoryParts.join('/')
```

I assume other people using Gitlab are experiencing this issue - it would be good to have more evidence of this though!

**Comment by CongeeCafe** [comment]
Yes I can confirm I'm having the same issue too. It would be nice to see it addressed.

**Comment by cpannwitz** [comment]
I'm having the same issue, though using a Github repository.

**Comment by TylerAHolden** [comment]
Same issue with a private github repo.

**Comment by Sly777** [comment]
Same issue with a private Github repo. The weird thing is that it works correctly for the first time today (which I implemented), but then it doesn't work anymore. I didn't make any even a change in the implementation.

**Comment by tysonmatanich** [comment]
Same issue with a private github repo. Worked for the first commit and the stopped working due to the 'SentryCLI failed to set commits.' error.

**Comment by tysonmatanich** [comment]
I was able to get it working, however the [article](https://blog.sentry.io/2020/06/08/automating-sentry-releases-with-our-netlify-build-plugin) by @jonesphillip needs updated to state that **private repos** need to enable GitHub integrations.

> Settings > Integrations > GitHub > Add Installation

After the integration was installed I then had to add the specific repo under the installed GitHub integrations under the 'Configurations' tab using the 'Add Repository' dropdown button.

**Comment by marcpicaud** [comment]
I had the same issue, this is how I solved it : 

> Settings > my-org > Integrations > GitHub

Then make sure your private repository is listed in the configuration. Then re-build on Netlify.

(Thanks @tysonmatanich for pointing me in the right direction.)

**Comment by dunsworth** [comment]
I have the same issue with Gitlab. It worked on the first run, but fails on subsequent runs with an "unknown repo" error. The Sentry UI recognizes the repo with no issue and the release that did work shows commits.

@djw27 What did you end up doing?

**Comment by djw27** [comment]
> I have the same issue with Gitlab. It worked on the first run, but fails on subsequent runs with an "unknown repo" error. The Sentry UI recognizes the repo with no issue and the release that did work shows commits.
> 
> @djw27 What did you end up doing?

@dunsworth we've left this as an unresolved issue on the back-burner - hopefully we'll get around coming up with a solution in the new year

**Comment by devs4522** [comment]
@djw27 Any update on this issue. I am facing same issue with Gilab integration. If it is not fixed any possible workaround will be much appreciated. Thanks

**Comment by djw27** [comment]
> @djw27 Any update on this issue. I am facing same issue with Gilab integration. If it is not fixed any possible workaround will be much appreciated. Thanks

@devs4522 no update yet I'm afraid - will post here when we get around to it

**Comment by juanpprieto** [comment]
I have the same issue with a github private repo. Worked on first try but fails on subsequent runs. What's odd is that the Configure github tab shows non of the repos I authorized via the dropdown. I've tried this multiple times and when I try to Add Installation the sentry github app is already authorized yet its not reflected on the sentry UI.

**Comment by jasikpark** [comment]
I think I might be having the same problem

**Comment by jasikpark** [comment]
Is it dependent on having your github integrated with sentry first??

**Comment by jonesphillip** [comment]
@jasikpark Yes you'll want to make sure you have your GitHub integration installed and the relevant repository added. You can find more information about that here: https://docs.sentry.io/product/releases/setup/release-automation/

Let me know if you're still having problems after that

**Comment by jonesphillip** [comment]
@juanpprieto did you add your add your repository as well? You'll need to do that as an additional step after installing the integration. More information on that here: https://docs.sentry.io/product/releases/setup/release-automation/

**Comment by VitalijusZimaitis** [comment]
Even tho SENTRY_REPOSITORY variable is set on Netlify its still not used when deploying

**Comment by alias-mac** [comment]
@VitalijusZimaitis that is because the version wasn't updated in Netlify plugins yet. See my comment here:
https://github.com/getsentry/sentry-netlify-build-plugin/pull/46#issuecomment-1038620658

**Comment by kasvith** [comment]
Im facing the same issue with GitLab

**Comment by kasvith** [comment]
For anyone still having this issue

1. Go to `Settings > Repositories`
2. Copy the name on the top (if your repo is `gitlab.com/xxx/yyy` then Sentry will show like `Xxx / Yyy`).
3. Go to Netlify > Build > Environment 
4. Create `SENTRY_REPOSITORY` and set this value you copied

Enjoy ✨

**Comment by saikojosh** [comment]
Life saver @kasvith! Spent way too long on this – perhaps the readme should be updated to indicate the name needs to exactly match the text string displayed in the Sentry UI and is not actually the repository slug.

**Comment by muh-hizbe** [comment]
i hace the same issue on private gitlab repo
